tests/UnifiedSpecTests/ManagesFailPointsTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace MongoDB\Tests\UnifiedSpecTests;
+
+use MongoDB\Driver\Server;
+use MongoDB\Operation\DatabaseCommand;
+use stdClass;
+
+use function PHPUnit\Framework\assertIsString;
+use function PHPUnit\Framework\assertObjectHasAttribute;
+
+trait ManagesFailPointsTrait
+{
+    /** @var list<list{string, Server}> */
+    private array $failPointsAndServers = [];
+
+    public function configureFailPoint(stdClass $failPoint, Server $server): void
+    {
+        assertObjectHasAttribute('configureFailPoint', $failPoint);
+        assertIsString($failPoint->configureFailPoint);
+        assertObjectHasAttribute('mode', $failPoint);
```

Comment:
A few assertions before we access object properties. I had these in for testing, but I suppose you can remove them if want to rely on the command succeeding.